examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/SafeMethodCachingInterceptor.java

```diff
@@ -234,14 +234,11 @@ public void onMessage(RespT message) {
 
               @Override
               public void onClose(Status status, Metadata trailers) {
-                if (cacheOptionsCausedCancellation) {
+                if (cacheOptionsErrorMsg != null) {
```

Comment:
this needs to be assigned to a local variable, as it could be different by the time you use it.

Reply:
It can only be set to a non-null value, so subsequent changes wouldn't really matter here (it will never go back to null). We'd want the latest value anyways.